contracts/prebuilts/account/utils/CrossChainTokenTransfer.sol

```diff
@@ -0,0 +1,311 @@
+    function allowlistDestinationChain(uint64 _destinationChainSelector, bool allowed) external onlyOwner {
+        allowlistedChains[_destinationChainSelector] = allowed;
+    }
+
+    function estimateLink(
+        uint64 _destinationChainSelector,
+        address _receiver,
+        address _token,
+        uint256 _amount
+    ) external view returns (uint estimate) {
+        // Create an EVM2AnyMessage struct in memory with necessary information for sending a cross-chain message
+        //  address(linkToken) means fees are paid in LINK
+        Client.EVM2AnyMessage memory evm2AnyMessage = _buildCCIPMessage(
+            _receiver,
+            _token,
+            _amount,
+            address(s_linkToken)
+        );
+
+        // Get the fee required to send the message
+        uint256 fees = s_router.getFee(_destinationChainSelector, evm2AnyMessage);
+
+        //Get 10% of the fees
+        uint256 tenPercent = fees.mul(10).div(100);
+        //Add 10% as slippage
+        estimate = fees.add(tenPercent);
+    }
+
+    /// @notice Transfer tokens to receiver on the destination chain.
+    /// @notice pay in LINK.
+    /// @notice the token must be in the list of supported tokens.
+    /// @notice This function can only be called by the owner.
+    /// @dev Assumes your contract has sufficient LINK tokens to pay for the fees.
+    /// @param _destinationChainSelector The identifier (aka selector) for the destination blockchain.
+    /// @param _receiver The address of the recipient on the destination blockchain.
+    /// @param _token token address.
+    /// @param _amount token amount.
+    /// @return messageId The ID of the message that was sent.
+    function transferTokensPayLINK(
+        uint64 _destinationChainSelector,
+        address _receiver,
+        address _sender,
+        address _token,
+        uint256 _amount,
+        uint256 _approvedAmountLink,
+        uint256 _approvedAmountToken
+    ) external onlyOwner onlyAllowlistedChain(_destinationChainSelector) returns (bytes32 messageId) {
+        // Create an EVM2AnyMessage struct in memory with necessary information for sending a cross-chain message
+        //  address(linkToken) means fees are paid in LINK
+        Client.EVM2AnyMessage memory evm2AnyMessage = _buildCCIPMessage(
+            _receiver,
+            _token,
+            _amount,
+            address(s_linkToken)
+        );
+
+        // Get the fee required to send the message
+        uint256 fees = s_router.getFee(_destinationChainSelector, evm2AnyMessage);
+
+        //verify amount approved for Link
+        if (_approvedAmountLink < fees) revert ApprovedAmountInsufficient(_approvedAmountLink, fees);
+
+        //verify amount approved for token transfered
+        if (_approvedAmountToken < _amount) revert ApprovedAmountInsufficient(_approvedAmountToken, fees);
+
+        //verify
+        if (fees > s_linkToken.balanceOf(address(this)))
```

Comment:
> This and the condition on line 120 can be combined. Or we can get rid of the condition on line 120 as we are just checking the parameter value against the fees, rather than the actual value which is being checked in this condition using `s_linkToken.balanceOf(address(this))`

This checks if our contract have enough token but I guess it can be removedn since if the approved amount is enough, we can get it from the user

Reply:
> Considering the Link & Transfer token approvals, their checks and withdrawals of respective tokens that are involved in this contract, we will have to deploy multiple instances of this contract, one for each account. You can deploy and instance of this from the createAccount(….) of BaseAccountFactory.sol while setting that account address as the owner of the CCIP contract instance.

Am not sure I understand you

Reply:
We shall discuss this over a call.